test/functional/metadata-builder/relation-join-column-builder/entity/City.ts

```diff
@@ -0,0 +1,14 @@
+import { Column, Entity, PrimaryColumn, Unique } from "../../../../../src"
+
+@Entity()
+@Unique(["id", "countryId"])
```

Comment:
Primary keys (in this case, `id`) already uniquely identify a row. So a unique constraint that includes the primary key is unnecessary.

Reply:
it will not be possible to create a foreign key constraint for Order#city

postgres example:
```
query failed: ALTER TABLE "order" ADD CONSTRAINT "FK_848d8fcd4aee3bd32043148f4bb" FOREIGN KEY ("cityId", "countryId") REFERENCES "city"("id","countryId") ON DELETE NO ACTION ON UPDATE NO ACTION
error: error: there is no unique constraint matching given keys for referenced table "city"
```

Reply:
Aaah, of course it is not possible when you have a simple primary key.

It was okay before, when the primary key was composite: `cityId and `countryId`.

Normally TypeORM creates the unique constraint automatically so you don't have to, unless there's already a unique index (in this case, the PK) that is a subset of it. Why doesn't TypeORM create it? Because some databases complain about this (because it doesn't make sense to them).

With a composite primary key (`countryName`, `cityName`), you could have:

City (`countryName`, `cityName`, `population`): `('France', 'Paris', 2100), ('Texas', 'Paris', 25)`